components/protocols/WebAuthnDemo.tsx

```
@@ -0,0 +1,1012 @@
+  // Simulated database
+  const [credentials, setCredentials] = useState<RegisteredCredential[]>([
+    {
+      id: 'cred_p256_mock_alice8a39b',
+      username: 'alice@cryptoviz.org',
+      publicKey: {
+        x: '8c4b1239fa8d2eb45a6c78bd9ef12c0a34b8c9d0e1f2a3b4c5d6e7f8a9b0c1d2',
+        y: '7d6e5c4b3a2a1f0e9d8c7b6a5a4b3c2d1e0f9e8d7c6b5a4a3b2c1d0e9f8e7d6c'
+      },
+      signCount: 14,
+      algorithm: 'ES256 (ECDSA P-256)',
+      createdAt: new Date(Date.now() - 3600000 * 24 * 5).toLocaleDateString(),
```

ESLint

[error] Error: Cannot call impure function during render

`Date.now` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

/home/jailuser/git/components/protocols/WebAuthnDemo.tsx:99:27
   97 |       signCount: 14,
   98 |       algorithm: 'ES256 (ECDSA P-256)',
>  99 |       createdAt: new Date(Date.now() - 3600000 * 24 * 5).toLocaleDateString(),
      |                           ^^^^^^^^^^ Cannot call impure function
  100 |       transports: ['internal']
  101 |     },
  102 |     {

(react-hooks/purity)